dotnet/src/SemanticKernel/SkillDefinition/SKFunctionExtensions.cs

```diff
@@ -92,48 +92,48 @@ public static ISKFunction UseFrequencyPenalty(this ISKFunction skFunction, doubl
     /// Execute a function allowing to pass the main input separately from the rest of the context.
     /// </summary>
     /// <param name="function">Function to execute</param>
+    /// <param name="kernel">Kernel</param>
     /// <param name="variables">Input variables for the function</param>
     /// <param name="skills">Skills that the function can access</param>
     /// <param name="culture">Culture to use for the function execution</param>
     /// <param name="settings">LLM completion settings (for semantic functions only)</param>
-    /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
     /// <returns>The result of the function execution</returns>
     public static Task<SKContext> InvokeAsync(this ISKFunction function,
+        IKernel kernel,
         ContextVariables? variables = null,
         IReadOnlySkillCollection? skills = null,
         CultureInfo? culture = null,
         CompleteRequestSettings? settings = null,
-        ILoggerFactory? loggerFactory = null,
         CancellationToken cancellationToken = default)
     {
-        var context = new SKContext(variables, skills, loggerFactory)
+        var context = new SKContext(kernel, variables, skills)
         {
             Culture = culture!
         };
 
-        return function.InvokeAsync(context, settings, cancellationToken);
+        return function.InvokeAsync(context, settings ?? function.RequestSettings, cancellationToken);
```

Comment:
Do we want to mark `InvokeAsync` as obsolete and direct people toward using `RunAsync` now?

Reply:
@markwallace-microsoft Not the scope of this PR yet. But we will as soon we can fully substitute all InvokeAsync cases.